$OpenBSD: patch-configure_ac,v 1.11 2020/02/08 10:36:01 cwen Exp $

Fix version string

libdl is not required on OpenBSD

Put port supplied CFLAGS on top priority so it builds with gcc by
supplying `-Wno-error' (upstreamed, can be removed for opensc > 0.20.0)

Index: configure.ac
--- configure.ac.orig
+++ configure.ac
@@ -363,29 +363,9 @@ elif test "${xslstylesheetsdir}" != "no"; then
 fi
 AC_MSG_RESULT([${xslstylesheetsdir}])
 
-AC_MSG_CHECKING([git checkout])
-GIT_CHECKOUT="no"
-if test -n "${GIT}" -a -d "${srcdir}/.git"; then
-	GIT_CHECKOUT="yes"
-fi
-AC_MSG_RESULT([${GIT_CHECKOUT}])
+OPENSC_SCM_REVISION="OpenSC-0.20.0, rev: 45e29056, commit-time: 2019-12-29 13:42:06 2019 +0100"
+OPENSC_VERSION_REVISION="0"
 
-if test "${GIT_CHECKOUT}" = "yes"; then
-	REVISION_DESCRIPTION="$(${GIT} describe || echo '<version not available>' )"
-	if test "${REVISION_DESCRIPTION}" = "<version not available>"; then
-		REVISION_DESCRIPTION="$(${GIT} describe --tags || echo '<version not available>')"
-	fi
-
-	HASH_COMMIT_DATE="$(${GIT} log -1 --pretty=format:'rev: %h, commit-time: %ci')"
-	GIT_TAG_COMMIT="$(${GIT} rev-list --tags --no-walk --max-count=1)"
-
-	OPENSC_SCM_REVISION="OpenSC-${REVISION_DESCRIPTION}, ${HASH_COMMIT_DATE}"
-	OPENSC_VERSION_REVISION="$(${GIT} rev-list ${GIT_TAG_COMMIT}..HEAD --count || echo 0)"
-else
-	OPENSC_SCM_REVISION="No Git revision info available"
-	OPENSC_VERSION_REVISION="0"
-fi
-
 dnl C Compiler features
 AC_C_INLINE
 
@@ -436,11 +416,6 @@ AC_CHECK_LIB(
 )
 
 if test "${WIN32}" = "no"; then
-	dnl dl support
-	AC_SEARCH_LIBS([dlopen], [dl dld], [], [
-		AC_MSG_ERROR([unable to find the dlopen() function])
-	])
-
 	dnl Special check for pthread support.
 	AX_PTHREAD(
 		[AC_DEFINE(
@@ -1061,7 +1036,6 @@ AM_CONDITIONAL([ENABLE_DNIE_UI], [test "${enable_dnie_
 AM_CONDITIONAL([ENABLE_NPATOOL], [test "${ENABLE_NPATOOL}" = "yes"])
 AM_CONDITIONAL([ENABLE_AUTOSTART], [test "${enable_autostart}" = "yes"])
 AM_CONDITIONAL([ENABLE_CMOCKA], [test "${enable_cmocka}" = "yes"])
-AM_CONDITIONAL([GIT_CHECKOUT], [test "${GIT_CHECKOUT}" = "yes"])
 AM_CONDITIONAL([ENABLE_FUZZING], [test "${enable_fuzzing}" = "yes"])
 AM_CONDITIONAL([ENABLE_SHARED], [test "${enable_shared}" = "yes"])
 AS_IF([test "${enable_shared}" = "yes"], [AC_DEFINE([ENABLE_SHARED], [1], [Enable shared libraries])])
@@ -1071,7 +1045,8 @@ if test "${enable_pedantic}" = "yes"; then
 	CFLAGS="${CFLAGS} -pedantic"
 fi
 if test "${enable_strict}" = "yes"; then
-	CFLAGS="${CFLAGS} -Wall -Wextra -Wno-unused-parameter -Werror"
+	# XXX This can be removed with opensc > 0.20.0 (upstreamed)
+	CFLAGS="-Wall -Wextra -Wno-unused-parameter -Werror ${CFLAGS}"
 fi
 
 AC_CONFIG_FILES([
